### ISSUE TITLE
Incorrect instruction_addrs in cfg node

### ISSUE THREAD

**Issue by HexRoman** [angr/angr#1481]
Given the following binary layout:
```

.text:000104B4                 EXPORT func_1
.text:000104B4 func_1
.text:000104B4                 STMFD   SP!, {R11,LR}
.text:000104B8                 ADD     R11, SP, #4
.text:000104BC                 MOV     R2, #0xC        ; n
.text:000104C0                 MOV     R1, #1          ; src
.text:000104C4                 LDR     R0, =asc_10590 
.text:000104C8                 BL      memcpy
.text:000104CC                 NOP
.text:000104D0                 LDMFD   SP!, {R11,PC}
.text:000104D0 ; End of function func_1
.text:000104D0 ; ---------------------------------------------------------------------------
.text:000104D4 dest            DCD asc_10590           ; DATA XREF: func_1+10↑r
.text:000104D8                 EXPORT func_2
.text:000104D8 func_2
.text:000104D8                 STMFD   SP!, {R11,LR}
.text:000104DC                 ADD     R11, SP, #4
.text:000104E0                 SUB     SP, SP, #8
.text:000104E4                 MOV     R0, #0x14       ; size
.text:000104E8                 BL      malloc
```

and the following code:
```
node = cfg.get_any_node(0x104D8, anyaddr=True)
print(node.size)
Out[1]: 24
print([hex(x) for x in node.instruction_addrs])
Out[2]: ['0x104d4']
```
It's seen that the instruction_addrs is incorrect, giving only the 'junk' instruction, at `0x104d4` and ignoring all other valid instructions in the node.

The file:
[example.zip](https://github.com/angr/angr/files/2983896/example.zip)

It happens in the latest version.

**Comment by rhelmot** [comment]
it is resolved. Thanks!

**Comment by HexRoman** [comment]
@rhelmot 
I have 2 problems after applying the fix:
- Sometimes new_node.instruction_addrs is tuple and other.instruction_addrs is list, and it can't add them together.
- Sometimes other.byte_string is None

What would you say about fixing it in the following way?

```
    def merge(self, other):
        """
        Merges this node with the other, returning a new node that spans the both.
        """
        new_node = self.copy()
        new_node.size += other.size
        new_node.instruction_addrs += tuple(other.instruction_addrs)
        new_node.byte_string += other.byte_string if other.byte_string else bytes()
        return new_node
```

**Comment by rhelmot** [comment]
there was another commit in the merge that included the above commit: aaf4544a01f9e30345f7309275c21a723a8e352b
